lib/rules/no-restricted-html-elements.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @author Doug Wade <douglas.b.wade@gmail.com>
+ */
+
+'use strict'
+
+const utils = require('../utils')
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'prevent usage of restricted html elements',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-restricted-html-elements.html'
+    },
+    fixable: null,
+    schema: {
+      type: 'array',
+      items: {
+        oneOf: [
+          { type: 'string' },
+          {
+            type: 'object',
+            properties: {
+              element: { type: 'string' },
+              message: { type: 'string', minLength: 1 }
+            },
+            required: ['element'],
+            additionalProperties: false
+          }
+        ]
+      },
+      uniqueItems: true,
+      minItems: 0
+    },
+    messages: {
+      restrictedHtmlElement: '{{message}}'
+    }
+  },
+  /**
+   * @param {RuleContext} context - The rule context.
+   * @returns {RuleListener} AST event handlers.
+   */
+  create(context) {
+    return utils.defineTemplateBodyVisitor(context, {
+      /**
+       * @param {VElement} node
+       */
+      VElement(node) {
+        if (!utils.isHtmlElementNode(node)) {
+          return
+        }
+
+        context.options.forEach((option) => {
+          const message =
+            option.message ||
+            `Unexpected use of forbidden HTML element ${node.rawName}.`
+          const element = option.element || option
+
+          if (element === node.rawName) {
+            context.report({
+              messageId: 'restrictedHtmlElement',
+              node: node.startTag,
+              data: { message }
+            })
```

Comment:
No need for the `restrictedHtmlElement` message ID placeholder in this case:

```suggestion
            context.report({
              message,
              node: node.startTag
            })
```

Reply:
This seems to be a way to prevent unintended behavior of placeholders when using custom messages. ESLint does the same.

https://github.com/eslint/eslint/blob/18f5e05bce10503186989d81ca484abb185a2c9d/lib/rules/no-restricted-syntax.js#L46